build_tools/integrate/stablehlo_tag_and_bump_version.sh

```diff
@@ -0,0 +1,213 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+exit_with_usage() {
+  echo "Usage: $0 [-t <COMMIT_TO_TAG>]"
+  echo "   -t  Specify a commit to tag, must be an integrated StableHLO commit"
+  echo "       available on https://github.com/search?q=repo%3Aopenxla%2Fxla+integrate+stablehlo&type=commits"
+  echo "       If not specifed, will only bump the 4w and 12w versions."
+  exit 1
+}
+
+## Tag old version
+tag_integrated_version() {
+  # Must be valid commit
+  if ! git rev-parse "$COMMIT" > /dev/null; then
+    echo "Could not find commit $COMMIT."
+    echo "Sync with upstream and pull changes to local repo."
+    exit 1
+  fi
+
+  # Ensure proper commit
+  # NOTE: THIS COMMIT MUST BE INTEGRATED INTO OPENXLA/XLA BY STABLEHLO ONCALL
+  # Valid commit hashes here:
+  # https://github.com/search?q=repo%3Aopenxla%2Fxla+integrate+stablehlo&type=commits
+  echo "Using commit:"
+  git log --format=%B -n 1 "$COMMIT"
+  echo
+  read -p "Is this the correct commit? [y] " -n 1 -r
+  echo; echo
+  if [[ "$REPLY" =~ ^[Nn]$ ]]; then
+    echo "Exiting..."
+    exit 1
+  fi
+
+  RELEASE="${VERSION[0]}.${VERSION[1]}.${VERSION[2]}"
+  RELEASE_TAG="v$RELEASE"
+  NEW_RELEASE="${VERSION[0]}.${VERSION[1]}.$((VERSION[2]+1))"
+  echo "Creating tagged release $RELEASE_TAG at $COMMIT"
+  echo "$ git tag -a $RELEASE_TAG $COMMIT -m \"StableHLO $RELEASE_TAG\""
+  git tag -a "$RELEASE_TAG" "$COMMIT" -m "StableHLO $RELEASE_TAG"
+  echo
+  echo "Most recent tags:"
+  git tag --sort=-version:refname | head -3
+  echo
+  echo "If this is incorrect, can undo using:"
+  echo "$ git tag -d $RELEASE_TAG"
+  echo
+}
+
+bump_patch_version() {
+  ## Bump revision
+  echo "Bumping revision to: $NEW_VERSION_STR"
+  echo "$ sed -i \"s/$OLD_VERSION_STR/$NEW_VERSION_STR/\" $VERSION_H"
+  sed -i "s/$OLD_VERSION_STR/$NEW_VERSION_STR/" "$VERSION_H"
+  echo
+}
+
+## Setup version variables as globals:
+##  VERSION, OLD_VERSION_STR, NEW_VERSION_STR
+SCRIPT_DIR="$(dirname "$(realpath "$0")")"
+VERSION_H="$SCRIPT_DIR/../../stablehlo/dialect/Version.h"
+VERSION_CPP="$SCRIPT_DIR/../../stablehlo/dialect/Version.cpp"
+setup_version_vars() {
+  # getCurrentVersion() { Version(0, X, Y); }
+  VERSION_STR=$(grep getCurrentVersion -A1 "$VERSION_H" | grep -o 'Version([0-9], .*)')
+  REGEX="Version\(([0-9]+), ([0-9]+), ([0-9]+)\)"
+  if [[ $VERSION_STR =~ $REGEX ]]
+  then
+    VERSION=("${BASH_REMATCH[1]}" "${BASH_REMATCH[2]}" "${BASH_REMATCH[3]}")
+    OLD_VERSION_STR="${BASH_REMATCH[0]}"
+    NEW_VERSION_STR="Version(${VERSION[0]}, ${VERSION[1]}, $((VERSION[2]+1)))"
+  else
+    echo "Error: Could not find current version string in Version.h" >&2
+    exit 1
+  fi
+}
+
+tag_and_bump() {
+  COMMIT="$1"
+  setup_version_vars
+  echo "Bumping version $OLD_VERSION_STR -> $NEW_VERSION_STR"
+
+  tag_integrated_version "$COMMIT"
```

Comment:
`tag_integrated_version` does not reed any parameters - it uses "external" $COMMIT var set inside `tag_and_bump`

Reply:
Yeah.. probably not very bash-y of me but I think it's easier to debug when functions are self contained. Any objection to this?

Reply:
If you prefer to make the function accept a parameter for modularity, you could adjust it to explicitly use the parameter instead of the global COMMIT variable

We can explicitly read arg0 to local variable COMMIT in both `tag_and_bump` and `tag_integrated_version`
```
local COMMIT="$1"
```
The local keyword is used to declare variables that are local to the function where they are defined. These variables are not accessible outside the function's scope

Reply:
Used local where relevant